server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -250,6 +252,45 @@ void dequeueShouldWorkAfterNetworkOutage() throws Exception {
                 .containsExactly(name1, name2, name3);
         }
 
+        @Test
+        void messagesShouldSurviveRabbitMQRestart() throws Exception {
+            String name1 = "myMail1";
+            String name2 = "myMail2";
+            String name3 = "myMail3";
+            Flux<MailQueue.MailQueueItem> dequeueFlux = Flux.from(getMailQueue().deQueue());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .build());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name2)
+                .build());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name3)
+                .build());
+
+            Thread.sleep(2000);
+            restartRabbitMQ();
+
+
+
+            List<MailQueue.MailQueueItem> items = dequeueFlux.take(3).collectList().block(Duration.ofSeconds(10));
+
+            assertThat(items)
+                .extracting(item -> item.getMail().getName())
+                .containsExactly(name1, name2, name3);
+        }
+
+        private void restartRabbitMQ() throws Exception {
```

Comment:
Shouldn't this be in rabbitMQExtension ?

Reply:
Yes, that's what i am thinking now that i work on the event bus PR.
Which lead me to the issue that the `DockerRabbitMQ.restart` method used in the `RabbitMQEventBustTest` do nothing. So all the tests pass but in reality nothing work as expected ... 
I will create an issue for this

Reply:
#3165